sipa/blueprints/usersuite.py

```diff
@@ -289,12 +283,9 @@ def change_mac():
         mac = form.mac.data
 
         try:
-            current_user.re_authenticate(password)
+            with current_user.tmp_authentication(password):
+                current_user.mac = mac
```

Comment:
Why use `tmp_authentication` when you don't actually need to access the password to change the mac? the only thing you are using from this context manager is the call to `re_authenticate`.

Why did you include the code to be executed on success in the `try`-block? “tell the logger things worked out” does not belong to the category „things that could go wrong“ (belongs in the `try`-block) but to „things that should be done if no exception was thrown“ (belongs in the `else`-block).

Reply:
I used `tmp_authentication` at this place mainly because it would have been the only line of code, except from tmp_authentication, directly calling `re_authenticate`. Besides, the upcoming pycroft datasource is going to need a password to change the MAC address.